### DISCUSSION TITLE
Error while define input types for action

### DISCUSSION THREAD

**Discussion by niekcandaele** [hasura/graphql-engine#6109] [Question] [answered]
Can anyone help me with creating an action please? Not sure whats going wrong here. Error message in comment at the bottom. Do I not create the UserInput type correctly?

![image](https://user-images.githubusercontent.com/22315101/97782394-97727280-1b91-11eb-8b88-888e12470e66.png)

**Comment by SusanAdey** [comment, marked as answer]
The new type definition should use input rather than type. You use type when defining the output from the action ie User. The following is what you want for the input type definition. You will need to define your output type structure as well. 

input UserInput {
   firstName: String!
   lastName: String!
   birthDate: timetz!
}
